src/dune_rules/cram/cram_rules.ml

```diff
@@ -49,53 +51,69 @@ let missing_run_t (error : Cram_test.t) =
 let test_rule
       ~sctx
       ~dir
-      ({ alias; loc; enabled_if; deps; locks; sandbox; packages = _ } : Spec.t)
+      ({ Spec.test_name_alias
+       ; extra_aliases
+       ; loc
+       ; enabled_if
+       ; deps
+       ; locks
+       ; sandbox
+       ; packages = _
+       } :
+        Spec.t)
       (test : (Cram_test.t, error) result)
   =
   let module Alias_rules = Simple_rules.Alias_rules in
-  let aliases = Alias.Name.Set.to_list_map alias ~f:(Alias.make ~dir) in
+  let extra_aliases =
+    Alias.Name.Set.remove extra_aliases test_name_alias
+    |> Alias.Name.Set.to_list_map ~f:(Alias.make ~dir)
+  in
+  let alias = Alias.make ~dir test_name_alias in
+  (* Here we get all other aliases to depend on the main alias which is the same as the
+     name of the cram test. *)
+  let* () =
+    Memo.parallel_iter extra_aliases ~f:(fun extra_alias ->
+      Rules.Produce.Alias.add_deps ~loc extra_alias (Action_builder.dep (Dep.alias alias)))
+  in
```

Comment:
Rather than use `Super_context.add_alias_action` I opted to directly register the dependency with `Rules.Produce.Alias.add_deps`.

Reply:
When I had to register an alias, I encounter these two options. What are the difference between them? My bet would be on the way they operate and that one call the other one, but I have always wondered if there was a behavior difference under the hood.